flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/UnboundedProcessingOverProcessFunction.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.table.runtime.aggregate
+
+import org.apache.flink.api.common.typeinfo.TypeInformation
+import org.apache.flink.configuration.Configuration
+import org.apache.flink.streaming.api.functions.ProcessFunction.{Context, OnTimerContext}
+import org.apache.flink.types.Row
+import org.apache.flink.streaming.api.functions.RichProcessFunction
+import org.apache.flink.util.{Collector, Preconditions}
+import org.apache.flink.api.common.typeutils.TypeSerializer
+import org.apache.flink.api.common.state.ValueStateDescriptor
+import org.apache.flink.api.java.typeutils.RowTypeInfo
+import org.apache.flink.api.common.state.ValueState
+import org.apache.flink.api.java.typeutils.runtime.RowSerializer
+
+class UnboundedProcessingOverProcessFunction(
+    private val aggregates: Array[Aggregate[_]],
+    private val projectionsMapping: Array[(Int, Int)],
+    private val aggregateMapping: Array[(Int, Int)],
+    private val  intermediateRowType: RowTypeInfo,
+  @transient private val returnType: TypeInformation[Row])
+  extends RichProcessFunction[Row, Row]{
+
+  protected var stateSerializer: TypeSerializer[Row] = _
+  protected var stateDescriptor: ValueStateDescriptor[Row] = _
+
+  private var output: Row = _
+  private var state: ValueState[Row] = _
+
+  override def open(config: Configuration) {
+    Preconditions.checkNotNull(aggregates)
+    Preconditions.checkNotNull(projectionsMapping)
+    Preconditions.checkNotNull(aggregateMapping)
+    Preconditions.checkArgument(aggregates.length == aggregateMapping.length)
+
+    val finalRowLength: Int = projectionsMapping.length + aggregateMapping.length
+    output = new Row(finalRowLength)
+    stateSerializer = intermediateRowType.createSerializer(getRuntimeContext.getExecutionConfig)
+    stateDescriptor = new ValueStateDescriptor[Row]("overState", stateSerializer)
```

Comment:
We can configure the state with a default value which is returned instead of `null` on the first access.